flow360/component/volume_mesh.py

```diff
@@ -772,3 +786,192 @@ def __getitem__(self, index) -> VolumeMesh:
     # pylint: disable=useless-parent-delegation
     def __iter__(self) -> Iterator[VolumeMesh]:
         return super().__iter__()
+
+
+###==== V2 API version ===###
+
+
+class VolumeMeshStatusV2(Enum):
+    """Status of volume mesh resource, the is_final method is overloaded"""
+
+    SUBMITTED = "submitted"
+    UPLOADING = "uploading"
+    UPLOADED = "uploaded"
+    COMPLETED = "completed"
+
+    def is_final(self):
+        """
+        Checks if status is final for volume mesh resource
+
+        Returns
+        -------
+        bool
+            True if status is final, False otherwise.
+        """
+        if self in [VolumeMeshStatusV2.COMPLETED]:
+            return True
+        return False
+
+
+class VolumeMeshMetaV2(AssetMetaBaseModelV2):
+    """
+    VolumeMeshMetaV2 component
+    """
+
+    project_id: str = pd.Field(alias="projectId")
+    deleted: bool = pd.Field()
+    status: VolumeMeshStatusV2 = pd.Field()  # Overshadowing to ensure correct is_final() method
```

Comment:
use pydantic v2 for new items

Reply:
Fixed in [6cece60](https://github.com/flexcompute/Flow360/pull/470/commits/6cece60781cc6aba8fc1ca2fdedd6b33711331be), also fixed usages in GeometryMeta for V2

Reply:
it doesn't look like pydantic v2 here?

Reply:
Fixed, mistook this comment for referring to NewVolumeMeshRequestV2 in requests.py where the exact same problem existed. My mistake.